torch/distributed/distributed_c10d.py

```diff
@@ -970,6 +970,13 @@ def _is_barrier_after_init() -> int:
     return int(os.getenv("TORCH_DIST_INIT_BARRIER", "0"))
 
 
+def _is_collective_hash_debug_enabled() -> int:
+    # Environment variable to control whether we hash the input/output of a collective
+    # so that we can verify that everything is working as expected at c10d level.
+    # `TORCH_NCCL_COLLECTIVE_HASH_DEBUG=1` to enable this validation.
+    return int(os.getenv("TORCH_NCCL_COLLECTIVE_HASH_DEBUG", "0"))
```

Comment:
curious if we absolutely need a new env var here. The main reason against a new one is there's already a couple env vars related to distributed debugging floating around, so we should form some principles on when to add new ones or use existing ones. For example, could this be integrated into torch_distributed_debug, or is it not a good fit?

Reply:
Agree that we don't want to add too many flags here. And we want to have some principles here indeed.

For this one specifically, we don't want it to be turned on even for DEBUG because it comes with extra cost? And what if we only want to see the hash value and don't want to see other detailed logs?